main/inc/lib/webservices/Rest.php

```diff
@@ -1590,4 +1591,59 @@ private function encodeParams(array $additionalParams = [])
 
         return $encoded;
     }
+
+    /**
+     * @param $params
+     *
+     * @throws Exception
+     *
+     * @return integer
+     */
+    public function createSessionFromModel($modelSessionId, $sessionName, $startDate, $endDate, array $extraFields = [])
+    {
+        if (!SessionManager::isValidId($modelSessionId)) {
+            throw new Exception(get_lang('ModelSessionDoesNotExist'));
+        }
+
+        $modelSession = SessionManager::fetch($modelSessionId);
+
+        foreach ($extraFields as $k => $v) {
+            if (array_key_exists($k, $modelSession)) {
+                $modelSession[$k] = $v;
+            }
+        }
+
+        if (api_is_multiple_url_enabled()) {
+            if (api_get_current_access_url_id() != -1) {
+                $modelSession['accessUrlId'] = api_get_current_access_url_id();
+            }
+        }
+
+        $newSessionId = SessionManager::create_session(
+            $sessionName,
+            $startDate,
+            $endDate,
+            $startDate,
+            $endDate,
+            $startDate,
+            $endDate,
+            $modelSession['coachId'],
+            $modelSession['sessionCategoryId'],
+            $modelSession['visibility'],
+            false,
+            $modelSession['duration'],
+            $modelSession['description'],
+            $modelSession['showDescription'],
+            $extraFields,
+            $modelSession['sessionAdminId'],
+            $modelSession['sendSubscriptionNotification'],
+            $modelSession['accessUrlId']
+        );
+
+        if (empty($newSessionId)) {
+            throw new Exception(get_lang('SessionNotRegistered'));
+        }
+
+        return [$newSessionId];
```

Comment:
The documentation of the function says it returns an integer. Here you are returning an array. Either change the documentation or the return value (I would prefer not returnin an array if you always will return a session ID (integer). Avoid unnecessary complexity, unless you are forced to do it because of some other code already in place.